WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -59,6 +65,38 @@ class MainActivity : AppCompatActivity() {
 
     }
 
+    /** Instantiate the Listener  */
+    class MyListener() : WebViewCompat.WebMessageListener {
+        override fun onPostMessage(
+            view: WebView,
+            message: WebMessageCompat,
+            sourceOrigin: Uri,
+            isMainFrame: Boolean,
+            replyProxy: JavaScriptReplyProxy
+        ) {
+            replyProxy.postMessage("Got it!")
+            Log.i("grcoleman", "received onPostMessage in app")
+        }
+    }
+
+    /** Instantiate a class to determine which JS-Java API to use based on the version the application is running on */
+    class JsObject() {
+        fun createJsObject(webview : WebView , context: Context) {
+            // The JavaScript object will be injected in any frame whose origin matches one in the list created below.
+            // We call the list rules because this is a set of allowed origin rules
+            val rules = setOf<String>("https://gcoleman799.github.io")
+            if (WebViewFeature.isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER)) WebViewCompat.addWebMessageListener(
+                webview,
+                "myObject",
```

Comment:
2 things:
1) Is there a better name for this interface than "myObject" that better describes what it does?
2) Could this name be extracted to a constant so that it can be used both here and below?

Reply:
For the first question, yes. I think jsObject might be a better name. Are you thinking of something with even more description?
For the second question, do you mean to create a string val in the MainActivity class that holds the value "jsObject"?

Reply:
Yes, I was thinking of something more descriptive of the methods it will provide? (something along the lines of `forecast` or `weather`... I'm not sure what makes the most sense)

For the second question, yes. Basically adding a `const val JS_OBJ_NAME = "jsObject"` to either the top level (above the class with a comment as what it is) or in a `companion object`, declared at the bottom of the class.